src/naming/Naming_AST.ml

```diff
@@ -963,5 +1002,9 @@ let resolve lang prog =
     end
   in
   visitor#visit_program () prog;
+
+  if lang_supports_implicit_return lang then
```

Comment:
It's better to do this in `Parse_target` after `Constant_propagation.propagate_dataflow` and preferably sharing the CFG. It may require a bit of refactoring (but should be a small one). We could sync and talk about this.

Reply:
Somehow it seems like you're commenting on a previous commit, but the feedback applies, and also thanks for the offline discussion to clarify.

Made some updates. See more comments below.